Upstream-Status: Submitted [https://gitlab.freedesktop.org/xorg/lib/libx11/merge_requests/33]
Signed-off-by: Ross Burton <ross.burton@intel.com>

From b41469c5f08b1f3365667ff3c430b104c9b8e25f Mon Sep 17 00:00:00 2001
From: Ross Burton <ross.burton@intel.com>
Date: Wed, 22 Jan 2020 17:11:23 +0000
Subject: [PATCH libX11] cpprules.in: squash whitespace in generated files

CPP is used to generate files, but as cpp reads files from the build host the
output has a number of blank lines at the beginning which varies depending on
what GCC and friends is used.

Pathalogical example:

 $ cpp -undef -traditional /dev/null
 # 1 "/dev/null"
 # 1 "<built-in>"
 # 1 "<command-line>"
 # 31 "<command-line>"
 # 1 "/usr/include/stdc-predef.h" 1 3 4

 # 17 "/usr/include/stdc-predef.h" 3 4

 [ 40 blank line ]

 # 32 "<command-line>" 2
 # 1 "/dev/null"

So depending on the content of stdc-predef.h and what other headers CPP will
load, the amount of whitespace in the generates files varies. This can result in
differences in reproducible environments, and file conflicts in multilib
environments.

As whitespace is irrelevant to these machine-readable files, extend the sed to
just delete blank lines.
---
 cpprules.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/cpprules.in b/cpprules.in
index 03d6701e..ce6b3c98 100644
--- a/cpprules.in
+++ b/cpprules.in
@@ -23,7 +23,8 @@ CPP_SED_MAGIC = $(SED) -e '/^\#  *[0-9][0-9]*  *.*$$/d' \
                        -e '/^[	 ]*XCOMM[^a-zA-Z0-9_]/s/XCOMM/\#/' \
                        -e '/^[	 ]*XHASH/s/XHASH/\#/' \
                        -e 's,X11_LOCALEDATADIR,$(X11_LOCALEDATADIR),g' \
-                       -e '/\@\@$$/s/\@\@$$/\\/'
+                       -e '/\@\@$$/s/\@\@$$/\\/' \
+                       -e '/^$$/d'
 
 .pre:
 	@$(MKDIR_P) $(@D)
-- 
2.20.1

